src/Bridge/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -240,16 +240,28 @@ protected function filterProperty(string $property, $value, QueryBuilder $queryB
         $association = $field;
         $valueParameter = $queryNameGenerator->generateParameterName($association);
 
-        $associationAlias = $this->addJoinOnce($queryBuilder, $queryNameGenerator, $alias, $association);
+        if ($metadata->isCollectionValuedAssociation($association)) {
+            $associationAlias = $this->addJoinOnce($queryBuilder, $queryNameGenerator, $alias, $association);
 
-        if (1 === count($values)) {
-            $queryBuilder
-                ->andWhere(sprintf('%s.id = :%s', $associationAlias, $valueParameter))
-                ->setParameter($valueParameter, $values[0]);
+            if (1 === count($values)) {
+                $queryBuilder
+                    ->andWhere(sprintf('%s.id = :%s', $associationAlias, $valueParameter))
+                    ->setParameter($valueParameter, $values[0]);
+            } else {
+                $queryBuilder
+                    ->andWhere(sprintf('%s.id IN (:%s)', $associationAlias, $valueParameter))
+                    ->setParameter($valueParameter, $values);
+            }
         } else {
-            $queryBuilder
-                ->andWhere(sprintf('%s.id IN (:%s)', $associationAlias, $valueParameter))
-                ->setParameter($valueParameter, $values);
+            if (1 === count($values)) {
+                $queryBuilder
+                    ->andWhere(sprintf('%s.%s = :%s', $alias, $association, $valueParameter))
+                    ->setParameter($valueParameter, $values[0]);
+            } else {
+                $queryBuilder
+                    ->andWhere(sprintf('%s.%s IN (:%s)', $alias, $association, $valueParameter))
+                    ->setParameter($valueParameter, $values);
+            }
```

Comment:
```php
$associationAlias = $metadata->isCollectionValuedAssociation($association) ? $this->addJoinOnce($queryBuilder, $queryNameGenerator, $alias, $association) : $association;
```

To avoid repeating the rest of the code twice.